### ISSUE TITLE
LearningRateScheduler callback: possibility to use schedule kwargs

### ISSUE THREAD

**Issue by vmarceau** [keras-team/keras#10625]
Currently, LearningRateScheduler parameter schedule must be a function that takes epoch index and learning rate as input arguments. When testing different learning rate annealing schemes, it is however sometimes useful to have a parametrizable schedule function that takes other various kwargs as inputs. 

Currently, one alternative to use the current  LearningRateScheduler callback with a parametrizable schedule function is to write a closure that returns a schedule with the appropriate input arguments (epoch, lr) and use the latter when calling the LearningRateScheduler constructor. 

In order to make the LearningRateScheduler callback more generic, I think it could be useful to be able to pass additional schedule kwargs to the LearningRateScheduler constructor, therefore allowing for more parameterizable schedule functions. 

If anybody would find that feature useful, I can write a PR for it.

Best regards,

VM

**Comment by brge17** [comment]
Not the best UX but a simple workaround is to use partial functions. 


```python
import math
from functools import partial

def _exp_policy(epoch, base_lr, gamma=0.1):
	""" Exponential Decay Policy. """
	return base_lr * math.pow(gamma, epoch)

for epoch in range(1, 10):
	print (_exp_policy(epoch, 2.0, gamma=0.5))

k_exp_policy = partial(_exp_policy, base_lr=2.0, gamma=0.5)
for epoch in range(1, 10):
	print(k_exp_policy(epoch))
```

**Comment by brge17** [comment]
The other rather glaring issue with the callback is that the most common policies: ``fixed``, ``step``, ``exponential`` etc... are not built in.

**Comment by vmarceau** [comment]
Hello @brge17,

Yes, partial functions is one workaround that makes use of python closures. Another one would be to use python decorators. As you said, this is not the best in terms of UX...

What I propose would be adding the schedule kwargs to the LearningRateScheduler function call. Something like this:
```
scheduler = keras.callbacks.LearningRateScheduler(schedule, verbose=0, **kwargs)
```

For example:
```
def schedule(epoch, lr, gamma):
	return lr * np.power(gamma, epoch)

scheduler = keras.callbacks.LearningRateScheduler(schedule, verbose=0, gamma=0.8)
```

A slightly different alternative would be to encapsulate the additional schedule kwargs in a config dictionary, like this:
```
scheduler = keras.callbacks.LearningRateScheduler(schedule, verbose=0, schedule_config={})
```

This would give:
```
def schedule(epoch, lr, gamma):
	return lr * np.power(gamma, epoch)

scheduler = keras.callbacks.LearningRateScheduler(schedule, verbose=0, schedule_config={'gamma': 0.8})
```

Please let me know if any of these suggestions make sense for you, and if so, I will be glad to send a PR for it.

Best regards,

**Comment by tRosenflanz** [comment]
Why not just write a custom callback that adjusts the learning rate based on your schedule? It is exceptionally simple to write custom callbacks

**Comment by vmarceau** [comment]
@tRosenflanz : I agree, thats another easy possibility. However, I thought having a parametrizable schedule function was a fairly general use case, and that therefore this improvement in the API could benefit many Keras users.

**Comment by vmarceau** [comment]
API Design Review submitted in https://github.com/keras-team/keras/issues/10644.
